src/renderer/navigation.ts

```diff
@@ -3,13 +3,24 @@
 import { ipcRenderer } from "electron";
 import { matchPath } from "react-router";
 import { reaction } from "mobx";
-import { createObservableHistory } from "mobx-observable-history";
-import { createBrowserHistory, createMemoryHistory, LocationDescriptor } from "history";
+import { createObservableHistory, IObservableHistory } from "mobx-observable-history";
+import { createBrowserHistory, createMemoryHistory, LocationDescriptor, History } from "history";
 import logger from "../main/logger";
 import { clusterViewRoute, IClusterViewRouteParams } from "./components/cluster-manager/cluster-view.route";
 
-export const history = typeof window !== "undefined" ? createBrowserHistory() : createMemoryHistory();
-export const navigation = createObservableHistory(history);
+let history: History
+let navigation: IObservableHistory
+
+if (ipcRenderer) {
+  history = typeof window !== "undefined" ? createBrowserHistory() : createMemoryHistory();
```

Comment:
Btw probably we don't need check for `window` anymore? Is is used in `main` somehow?

Reply:
+ remove `createMemoryHistory()`

Reply:
Right, I'll cleanup more.

Reply:
Done.